lib/mix/lib/mix/tasks/escript.build.ex

```diff
@@ -242,6 +247,19 @@ defmodule Mix.Tasks.Escript.Build do
     end)
   end
 
+  defp strip_beams(tuples) do
+    for {basename, maybe_beam} <- tuples do
+      {basename, strip_beam(maybe_beam)}
+    end
+  end
+
+  defp strip_beam(maybe_beam) do
+    case :beam_lib.strip(maybe_beam) do
+      {:ok, {_, stripped_beam}} -> stripped_beam
+      {:error, :beam_lib, {:not_a_beam_file, _}} -> maybe_beam
```

Comment:
It still could fail if beam file has missing chunk or for some other reason (permissions). 🤔